File: safedialog/src/main/java/sgtmelon/safedialog/library/OptionsDialog.java
package sgtmelon.safedialog.library;

import android.app.Dialog;
import android.content.Context;
import android.content.DialogInterface;
import android.os.Bundle;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.appcompat.app.AlertDialog;
import androidx.fragment.app.DialogFragment;
import sgtmelon.safedialog.office.annot.DialogAnn;

public final class OptionsDialog extends DialogFragment implements DialogInterface.OnClickListener {

    private Context context;

    private String[] items;
    private int position;

    private DialogInterface.OnClickListener onClickListener;

    public void setArguments(String[] items, int p) {
        final Bundle bundle = new Bundle();

        bundle.putStringArray(DialogAnn.INIT, items);
        bundle.putInt(DialogAnn.VALUE, p);

        setArguments(bundle);
    }

    @Override
    public void onAttach(Context context) {
        super.onAttach(context);
        this.context = context;
    }

    @NonNull
    @Override
    public Dialog onCreateDialog(@Nullable Bundle savedInstanceState) {
        final Bundle bundle = getArguments();

        if (savedInstanceState != null) {
            items = savedInstanceState.getStringArray(DialogAnn.INIT);
            position = savedInstanceState.getInt(DialogAnn.VALUE);
        } else if (bundle != null) {
            items = bundle.getStringArray(DialogAnn.INIT);
            position = bundle.getInt(DialogAnn.VALUE);
        }

        return new AlertDialog.Builder(context)
                .setItems(items, this)
                .setCancelable(true)
                .create();
    }

    @Override
    public void onSaveInstanceState(@NonNull Bundle outState) {
        super.onSaveInstanceState(outState);

        outState.putStringArray(DialogAnn.INIT, items);
        outState.putInt(DialogAnn.VALUE, position);
    }

    public void setOnClickListener(DialogInterface.OnClickListener onClickListener) {
        this.onClickListener = onClickListener;
    }

    public int getPosition() {
        return position;
    }

    @Override
    public void onClick(DialogInterface dialogInterface, int i) {
        if (onClickListener != null) onClickListener.onClick(dialogInterface, i);
    }

}

File: safedialog/src/main/java/sgtmelon/safedialog/library/OptionsDialog.kt
package sgtmelon.safedialog.library

import android.app.Dialog
import android.content.DialogInterface
import android.os.Bundle
import androidx.appcompat.app.AlertDialog
import androidx.fragment.app.DialogFragment
import sgtmelon.safedialog.office.annot.DialogAnn

class OptionsDialog : DialogFragment(), DialogInterface.OnClickListener {

    lateinit var onClickListener: DialogInterface.OnClickListener

    private lateinit var items: List<String>

    var position: Int = 0
        private set

    fun setArguments(items: Array<String>, p: Int) {
        val bundle = Bundle()

        bundle.putStringArray(DialogAnn.INIT, items)
        bundle.putInt(DialogAnn.VALUE, p)

        arguments = bundle
    }

    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
        val bundle = arguments

        items = savedInstanceState?.getStringArray(DialogAnn.INIT)?.toList()
                ?: bundle?.getStringArray(DialogAnn.INIT)?.toList()
                ?: ArrayList()

        position = savedInstanceState?.getInt(DialogAnn.VALUE)
                ?: bundle?.getInt(DialogAnn.VALUE)
                ?: 0

        return AlertDialog.Builder(context!!)
                .setItems(items.toTypedArray(), this)
                .setCancelable(true)
                .create()
    }

    override fun onSaveInstanceState(outState: Bundle) {
        super.onSaveInstanceState(outState)

        outState.putStringArray(DialogAnn.INIT, items.toTypedArray())
        outState.putInt(DialogAnn.VALUE, position)
    }

    override fun onClick(dialogInterface: DialogInterface, i: Int) {
        onClickListener.onClick(dialogInterface, i)
    }

}